example/advection/advection_package.cpp

```diff
@@ -302,6 +317,25 @@ void PreFill(MeshBlockData<Real> *rc) {
           v(out + n, k, j, i) = 1.0 - v(in + n, k, j, i);
         });
   }
+
+  // Fill the metadata::None var with index gymnastics.
+  if (test_metadata_none) {
+    IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+    IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+    IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
+
+    // packing in principle unnecessary/convoluted here and just done for demonstration
+    std::vector<std::string> vars({"metadata_none_var"});
+    PackIndexMap imap;
+    const auto &v = rc->PackVariables(vars, imap);
+
+    const int ivar = imap.get("metadata_none_var").first;
+    pmb->par_for(
+        PARTHENON_AUTO_LABEL, 0, 2, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+        KOKKOS_LAMBDA(const int n, const int k, const int j, const int i) {
+          v(ivar, n, k, j, i) = n + k * j * i;
+        });
+  }
```

Comment:
I'm not sure what/why this is happening here.
If I read the the field correctly, then it's a vector with 3 components and extents larger than the mesh size by one.
In this `par_for` the block indices are used (without any global offset), so every block will have the same indices set to the same values and a large fraction of the index space will remain 0.
Is that intentional?

Reply:
Good catch, I didn't pay much mind to the contents of the fields. This should be fixed to be filled now, I think.